packages/evm/src/block.ts

```diff
@@ -1,361 +1,365 @@
-import { Bytes, BytesFromUint8Array } from "@apibara/protocol";
-import { Schema } from "@effect/schema";
-
+import { BytesFromUint8Array } from "@apibara/protocol";
+
+import {
+  ArrayCodec,
+  BigIntCodec,
+  BooleanCodec,
+  type Codec,
+  type CodecType,
+  DateCodec,
+  MessageCodec,
+  NumberCodec,
+  OneOfCodec,
+  OptionalCodec,
+  StringCodec,
+} from "@apibara/protocol/codec";
+import { RequiredCodec } from "@apibara/protocol/codec";
 import { Address, B256, U128, U256 } from "./common";
-import { tag } from "./helpers";
 import * as proto from "./proto";
 
-export const Bloom = Schema.transform(
-  Schema.Struct({
-    value: BytesFromUint8Array,
-  }),
-  Bytes,
-  {
-    strict: false,
-    encode(value) {
-      throw new Error("Not implemented");
-    },
-    decode({ value }) {
-      return value;
-    },
+export const Bloom: Codec<
+  Address | undefined,
```

Comment:
This type should be `0x${string}`. I know it's the same, but reading the current type is confusing.